src/foam/nanos/rope/ROPEAuthorizer.java

```diff
@@ -106,29 +106,28 @@ public boolean ropeSearch(ROPEActions operation, FObject obj, X x, String target
   }
 
   public List<FObject> getSourceObjects(X x, ROPE rope, FObject obj) {
-    DAO junctionDAO = (DAO) x.get(rope.getJunctionDAOKey());
-    DAO sourceDAO = (DAO) x.get(rope.getSourceDAOKey());
     List<FObject> sourceObjs = new ArrayList(); 
 
     if ( rope.getCardinality().equals("*:*") ) {
 
-      Object predicateProperty = rope.getIsInverse() ? rope.getJunctionModel().getAxiomByName("sourceId") : rope.getJunctionModel().getAxiomByName("targetId");
-      List<FObject> junctionObjs = ((ArraySink) junctionDAO
-        .where(
-          EQ(predicateProperty, (Long) retrieveProperty(obj, "get", "id"))
-        )
-        .select(new ArraySink()))
-        .getArray();
+      foam.dao.ManyToManyRelationshipImpl relationship = (foam.dao.ManyToManyRelationshipImpl) retrieveProperty(obj, "get", rope.getRelationshipKey(), x);
+      List<FObject> junctionObjs = (List<FObject>) ( (ArraySink) relationship.getJunctionDAO().where(
+        EQ(relationship.getJunction().SOURCE_ID, (Long) retrieveProperty(obj, "get", "id"))
```

Comment:
syntax error in this one in Mlang query 
refer to this for the first arg in EQ()
```Object predicateProperty = rope.getIsInverse() ? rope.getJunctionModel().getAxiomByName("sourceId") : rope.getJunctionModel().getAxiomByName("targetId");```

Reply:
try again